azkaban-web-server/src/restli/java/azkaban/restli/ResourceUtils.java

```diff
@@ -56,4 +59,30 @@ public static User getUserFromSessionId(String sessionId, String ip)
 
     return session.getUser();
   }
+
+  public static String getRealClientIpAddr(ResourceContext context){
+
+    // If some upstream device added an X-Forwarded-For header
+    // use it for the client ip
+    // This will support scenarios where load balancers or gateways
+    // front the Azkaban web server and a changing Ip address invalidates
+    // the session
+    Map<String, String> headers = context.getRequestHeaders();
+
+    String clientIp = headers.getOrDefault("X-Forwarded-For", null);
+    if(clientIp == null){
+      clientIp = (String) context.getRawRequestContext().getLocalAttr("REMOTE_ADDR");
+    }
+    else{
```

Comment:
Could you think of a way to reduce the code logic duplication in this method and the other one?